src/main/java/com/WebController/EventController.java

```diff
@@ -65,4 +33,37 @@
         return events;
     }
 
+    @PostMapping (value = "/eventinfo")
+    public Patient addEvent (@RequestBody Event event, HttpSession session, Principal principal) {
+        //prevent user from input garbage event with empty name, empty date or empty place.
+        if (event.getDate()==null ||event.getName().equals("unknwon") || event.getName().equals("") || event.getPlace().getAddress().equals("") || event.getPlace().getAddress().equals("unknown")){
+            return null;
+        }
+        boolean isVerified = false;
+        if (principal!=null && this.userRepository.findUserByEmailAddressIgnoreCase(principal.getName())!=null){
+            isVerified = true;
+        }
+        Optional<Patient> p = this.patientRepository.findById(session.getId());
+        Patient patient;
+        if (p.isPresent()){
+            patient = p.get();
+        } else {
+            patient = new Patient(session.getId());
+        }
+        Event existEvent =  this.eventRepo.findAllByNameAndDate(event.getName(), event.getDate());
```

Comment:
is name and date sufficient to guarantee uniqueness of the event? what if the same event happens in multiple locations? this is why equals() should be somehow used here...

Reply:
Agree. I will change the method to findAllByNameAndDate () to get a list. 
Then use equals() method for the list.